### PR TITLE
CI-runnable tests for violin plots (SCP-2213, SCP-2391, SCP-2403)

### PR DESCRIPTION
**Pull request by eweitz**
This adds CI-runnable tests for violin plots.  It introduces a general model for fast, reliable, and higher-coverage tests of React components that use Plotly.

Previously, it was impractical to have automated tests for components that use Plotly.  Existing end-to-end tests in `ui_test_suite.rb` have  high false-positive rates -- e.g. throwing an error when no bug exists -- and are not runnable in Travis CI due to authentication issues.  Also, code in `useEffect` and downstream was not testable at a component level.

Now, tests for such components can run in CI by mocking `fetch` with a file-system read (rather than an HTTP request) using high-fidelity mock data.  Using React Testing Library (originally researched by Eno), we can now test code in `useEffect` -- and thus have reliable regression tests for issues like #560.  These tests are also fast; this one typically takes < 100 ms.

The main limitation here is that it does not actually test rendered SVG, due to jsdom/jsdom#1330.  Something like [`jest-puppeteer`](https://github.com/smooth-code/jest-puppeteer) could enable that in a similarly fast, reliable, and Travis CI-friendly manner.

This satisfies SCP-2213, SCP-2391, and SCP-2403.

### CODE REVIEW

**Comment by eweitz** [comment]
This PR includes changes from #572, and will get a simpler to review once that is merged.

File: rails_local.bash [hunk 1]
```diff
@@ -0,0 +1,5 @@
+#!/bin/bash
+
+source config/secrets/.source_env.bash
+
+thin start --ssl --ssl-key-file config/local_ssl/localhost.key --ssl-cert-file config/local_ssl/localhost.crt  --ssl-disable-verify
```
Comment (bistline) [inline rails_local.bash:5]: Since we're using `puma` now, is this still needed?

Comment (eweitz) [reply rails_local.bash]: Nope, good catch, removed in 2657056.

File: config/local_ssl/localhost.crt [hunk 2]
```diff
@@ -0,0 +1,28 @@
+-----BEGIN CERTIFICATE-----
```
Comment (devonbush) [inline config/local_ssl/localhost.crt:1]: this file should be removed -- it got moved to config/certs

File: config/local_ssl/localhost.key [hunk 3]
```diff
@@ -0,0 +1,52 @@
+-----BEGIN PRIVATE KEY-----
```
Comment (devonbush) [inline config/local_ssl/localhost.key:1]: same with this file -- moved to config/certs

File: config/local_ssl/localhost.crt (hunk 2 above)
Comment (eweitz) [reply config/local_ssl/localhost.crt]: Done in 4ee6b23.

File: config/local_ssl/localhost.key (hunk 3 above)
Comment (eweitz) [reply config/local_ssl/localhost.key]: Done in 4ee6b23.

**Pull request merged**